src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java

```diff
@@ -392,13 +400,15 @@ public Flow updateFlow(FlowOperationsCarrier carrier, FlowPatch flowPatch)
 
         Flow updatedFlow = updateFlowResult.getUpdatedFlow();
         if (updateFlowResult.isNeedUpdateFlow()) {
+            saveHistoryActionFullUpdate(carrier, updatedFlow, flowPatch);
```

Comment:
will we create 2 events in history? one here and one in flowhs topology?

Reply:
yes. And I think it make sense because this is actually what happens. The patch event would contain in this case that it's been invoked and then it passed the control to the update. The update history will follow.